src/waveform/waveformwidgetfactory.cpp

```diff
@@ -137,6 +138,11 @@ WaveformWidgetFactory::WaveformWidgetFactory() :
         glFormat.setRgba(true);
         QGLFormat::setDefaultFormat(glFormat);
 
+        std::unique_ptr<QGLWidget> pGlWidget = std::make_unique<QGLWidget>(); // create paint device
```

Comment:
QScopedPointer would be more appropriate here, because we don't want to let the widget live longer than the current scope.